crates/accelerate/src/synthesis/linear/pmh.rs

```diff
@@ -0,0 +1,195 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+use ndarray::{s, Array1, Array2, ArrayViewMut2, Axis};
+use numpy::PyReadonlyArray2;
+use smallvec::smallvec;
+use std::cmp;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+use pyo3::prelude::*;
+
+use super::utils::_add_row_or_col;
+
+/// This helper function allows transposed access to a matrix.
+fn _index(transpose: &bool, i: &usize, j: &usize) -> (usize, usize) {
+    if *transpose {
```

Comment:
Is there a reason to use `&bool`, `&usize` instead of `bool`, `usize`? I don't think we get any memory saving by passing the arguments "by reference" rather than directly.

Reply:
Nope, just initial Rust hiccups 😉 I'll change to use by value (same for your comment below).